### PR TITLE
Feature/lib 29

### PR DESCRIPTION
**Pull request by Emilo067**
# Pull Request Description

## Overview

This pull request introduces a new `Tabs` component along with its related subcomponents (`TabsList`, `TabsTrigger`, `TabsContent`). These components are built using the Radix UI's `Tabs` primitives, providing a flexible and accessible tabbed interface with custom styling.

## Changes

- **New Components**:
  - `Tabs`: The main component that wraps around Radix UI's `Tabs.Root` and applies custom styles.
  - `TabsList`: A container component for grouping `TabsTrigger` elements.
  - `TabsTrigger`: A clickable tab trigger that changes the visible tab content.
  - `TabsContent`: A container for the content associated with a specific tab.
  
- **Styling**:
  - Added custom styles in `tabs.module.scss` to style the `Tabs`, `TabsList`, `TabsTrigger`, and `TabsContent` components.
  - The styles include focus states, hover states, active states, and disabled states for the `TabsTrigger` component.

## Testing

- Unit tests have been added for the new `Tabs` components.
- Visual testing has been performed to ensure the correct application of styles.
- Interaction tests have been included to verify the behavior of tab switching and state changes.

## Documentation

- Added JSDoc comments to all new components to document their props and behavior.
- Updated the storybook stories to include examples of the `Tabs` component in various states (default, disabled, with error states).

## Notes

- The components are designed to be easily extendable with additional styling or behavior modifications.
- Future enhancements could include additional accessibility features or custom animations.

---

Please review the changes and provide feedback or approval for merging into the `dev` branch.

### CODE REVIEW

File: src/components/ui/tabs/tabs.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+ * @param {React.ReactNode} props.children - The tab triggers and contents.
+ * @param {string} [props.className] - Additional class name for styling the component.
+ * @param {function} [props.onValueChange] - Callback function that is triggered when the active tab value changes.
+ * @param {React.Ref<HTMLDivElement>} ref - Reference to the underlying Radix Tabs Root element.
+ * @returns {JSX.Element} The Tabs component.
+ */
+
+export const Tabs = forwardRef<ElementRef<typeof TabsRadix.Root>, TabsProps>(
+  ({ onValueChange, children, className, ...rest }, ref) => {
+    const classNames = clsx(styles.root, className)
```
Comment (Lelik7777) [inline src/components/ui/tabs/tabs.tsx]: сделай через {root}as const

File: src/components/ui/tabs/tabs.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+ * @component
+ * @param {Object} props - The component props.
+ * @param {string} props.value - The value that will be assigned to this trigger. This value will be passed to onValueChange when the tab is selected.
+ * @param {React.Ref<HTMLButtonElement>} ref - Reference to the underlying Radix Tabs Trigger element.
+ * @returns {JSX.Element} The TabsTrigger component.
+ */
+
+export const TabsTrigger = forwardRef<ElementRef<typeof TabsRadix.Trigger>, TabsTriggerProps>(
+  ({ value, ...rest }, ref) => {
+    return <TabsRadix.Trigger value={value} className={styles.trigger} ref={ref} {...rest} />
```
Comment (Lelik7777) [inline src/components/ui/tabs/tabs.tsx]: для унификации создай объект classNames

File: src/components/ui/tabs/tabs.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+ * @param {Object} props - The component props.
+ * @param {React.ReactNode} props.children - The content to be displayed when the associated tab is active.
+ * @param {React.Ref<HTMLDivElement>} ref - Reference to the underlying Radix Tabs Content element.
+ * @returns {JSX.Element} The TabsContent component.
+ */
+
+export const TabsContent = forwardRef<ElementRef<typeof TabsRadix.Content>, TabsContentProps>(
+  ({ children, ...rest }, ref) => {
+    return (
+      <TabsRadix.Content className={styles.content} ref={ref} {...rest}>
```
Comment (Lelik7777) [inline src/components/ui/tabs/tabs.tsx]: для унификации создай объект classNames

File: src/components/ui/tabs/tabs.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        {children}
+      </TabsRadix.Content>
+    )
+  }
+)
```
Comment (Lelik7777) [inline src/components/ui/tabs/tabs.tsx:105]: add for debugging

Tabs.displayName = 'Tabs'
TabsList.displayName = 'TabsList'
TabsTrigger.displayName = 'TabsTrigger'

File: src/components/ui/tabs/tabs.tsx (hunk 1 above)
Comment (Emilo067) [reply src/components/ui/tabs/tabs.tsx]: исправил

File: src/components/ui/tabs/tabs.tsx (hunk 3 above)
Comment (Emilo067) [reply src/components/ui/tabs/tabs.tsx]: исправил

File: src/components/ui/tabs/tabs.tsx (hunk 4 above)
Comment (Emilo067) [reply src/components/ui/tabs/tabs.tsx:105]: исправил

File: src/components/ui/tabs/tabs.tsx [hunk 5]
```diff
@@ -67,8 +69,11 @@ export type TabsTriggerProps = ComponentPropsWithoutRef<typeof TabsRadix.Trigger
  */
 
 export const TabsTrigger = forwardRef<ElementRef<typeof TabsRadix.Trigger>, TabsTriggerProps>(
-  ({ value, ...rest }, ref) => {
-    return <TabsRadix.Trigger value={value} className={styles.trigger} ref={ref} {...rest} />
+  ({ value, className, ...rest }, ref) => {
+      const classNames = {
+          trigger: clsx(styles.trigger, className)
+      }
```
Comment (Lelik7777) [inline src/components/ui/tabs/tabs.tsx]: as const

File: src/components/ui/tabs/tabs.tsx [hunk 6]
```diff
@@ -86,11 +91,19 @@ export type TabsContentProps = ComponentPropsWithoutRef<typeof TabsRadix.Content
  */
 
 export const TabsContent = forwardRef<ElementRef<typeof TabsRadix.Content>, TabsContentProps>(
-  ({ children, ...rest }, ref) => {
-    return (
-      <TabsRadix.Content className={styles.content} ref={ref} {...rest}>
+  ({ children, className,  ...rest }, ref) => {
+        const classNames = {
+            content: clsx(styles.content, className)
+        }
```
Comment (Lelik7777) [inline src/components/ui/tabs/tabs.tsx]: as const

**Review by NoName-ForTeam — APPROVED** [review]

**Review by EvgeniyaYakunina — APPROVED** [review]

File: src/components/ui/tabs/tabs.tsx (hunk 5 above)
Comment (Emilo067) [reply src/components/ui/tabs/tabs.tsx]: fix

File: src/components/ui/tabs/tabs.tsx (hunk 6 above)
Comment (Emilo067) [reply src/components/ui/tabs/tabs.tsx]: fix

**Pull request merged**